app/controllers/api/v2/providers_controller.rb

```diff
@@ -79,18 +79,24 @@ def suggest_any
         return render(status: :bad_request) if params[:query].nil? || params[:query].length < 2
         return render(status: :bad_request) unless begins_with_alphanumeric(params[:query])
 
-        found_providers = @recruitment_cycle.providers
-          .search_by_code_or_name(params[:query])
-          .limit(5)
+        scope = @recruitment_cycle.providers
+                                  .search_by_code_or_name(params[:query])
+                                  .limit(5)
+
+        scope = scope.are_accredited_bodies if accredited_body_filter?
 
         render(
-          jsonapi: found_providers,
+          jsonapi: scope,
           class: { Provider: SerializableProviderSuggestion },
         )
       end
 
     private
 
+      def accredited_body_filter?
+        params[:filter] && params[:filter][:accredited_body] == "true"
```

Comment:
```suggestion
        params.dig(:filter, :only_accredited_body) == "true"
```

`accredited_body` I'm not sure of the naming maybe `only_accredited_body` since it a tristate


>     should only return accredited bodies